File: blab-api/src/data-access/user-queries.ts
import { DbFactory } from './db-factory';
import { User } from '../shared/models/user';

export class UserQueries {
  async getUser(userName: string): Promise<User> {
    const db = await DbFactory.get();
    const user = await db
      .collection<User>('users')
      .findOne({ userName: userName });
    return user;
  }
}

File: blab-api/internal/data-access/user-queries.go
package dataaccess

import (
	"blab/api/internal/database"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type UserQueries struct {
}

type User struct {
	ID       primitive.ObjectID
	Email    string
	UserName string
}

func (u UserQueries) Get() (User, error) {
	collection := database.Client.Database("blab").Collection("users")
	var user User
	err := collection.FindOne(database.Ctx, bson.M{"email": "test@test.com"}).Decode(&user)
	if err != nil {
		return user, err
	}
	return user, nil
}
